concepts/optional-type/links.json

```diff
@@ -0,0 +1,2 @@
+[]
```

Comment:
Suggest a link to the Javadocs for Optional. See the links.json for the [Lists](https://github.com/exercism/java/blob/main/concepts/lists/links.json) concept for an example.